tianshou/policy/base.py

```diff
@@ -261,8 +261,12 @@ def map_action_inverse(
 
     def process_buffer(
         self,
-        buffer: ReplayBuffer | None,
-    ) -> ReplayBuffer | None:
+        buffer: ReplayBuffer,
+    ) -> ReplayBuffer:
```

Comment:
A more correct annotation would be

```python
TBuffer = TypeVar("TBuffer", bound=ReplayBuffer)

    ...
    def process_buffer(self, buffer: TBuffer) -> TBuffer:
       ...
```

You don't have to do it now, if you don't want to, but I think it'd be nice :)